SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -29,52 +29,62 @@ ComponentReader::ComponentReader(USpatialNetDriver* InNetDriver, FObjectReferenc
 
 void ComponentReader::ApplyComponentData(const Worker_ComponentData& ComponentData, UObject* Object, USpatialActorChannel* Channel, bool bIsHandover)
 {
-	Schema_Object* ComponentObject = Schema_GetComponentDataFields(ComponentData.schema_type);
-
-	if (bIsHandover)
+	if (Object->IsPendingKill())
 	{
-		ApplyHandoverSchemaObject(ComponentObject, Object, Channel, true);
-	}
-	else
-	{
-		ApplySchemaObject(ComponentObject, Object, Channel, true);
+		return;
 	}
-}
 
-void ComponentReader::ApplyComponentUpdate(const Worker_ComponentUpdate& ComponentUpdate, UObject* Object, USpatialActorChannel* Channel, bool bIsHandover)
-{
-	Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(ComponentUpdate.schema_type);
+	Schema_Object* ComponentObject = Schema_GetComponentDataFields(ComponentData.schema_type);
 
 	TArray<Schema_FieldId> ClearedIds;
-	ClearedIds.SetNum(Schema_GetComponentUpdateClearedFieldCount(ComponentUpdate.schema_type));
-	Schema_GetComponentUpdateClearedFieldList(ComponentUpdate.schema_type, ClearedIds.GetData());
+	TArray<uint32> UpdatedIds;
+	UpdatedIds.SetNumUninitialized(Schema_GetUniqueFieldIdCount(ComponentObject));
+	Schema_GetUniqueFieldIds(ComponentObject, UpdatedIds.GetData());
 
 	if (bIsHandover)
 	{
-		ApplyHandoverSchemaObject(ComponentObject, Object, Channel, false, &ClearedIds);
+		ApplyHandoverSchemaObject(ComponentObject, Object, Channel, true, UpdatedIds, ClearedIds);
 	}
 	else
 	{
-		ApplySchemaObject(ComponentObject, Object, Channel, false, &ClearedIds);
+		ApplySchemaObject(ComponentObject, Object, Channel, true, UpdatedIds, ClearedIds);
 	}
 }
 
-void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject* Object, USpatialActorChannel* Channel, bool bIsInitialData, TArray<Schema_FieldId>* ClearedIds)
+void ComponentReader::ApplyComponentUpdate(const Worker_ComponentUpdate& ComponentUpdate, UObject* Object, USpatialActorChannel* Channel, bool bIsHandover)
 {
-	TArray<uint32> UpdateFields;
-	UpdateFields.SetNum(Schema_GetUniqueFieldIdCount(ComponentObject));
-	Schema_GetUniqueFieldIds(ComponentObject, UpdateFields.GetData());
-
-	if (UpdateFields.Num() == 0)
+	if (Object->IsPendingKill())
 	{
 		return;
 	}
 
-	if(Object->IsPendingKill())
+	Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(ComponentUpdate.schema_type);
+
+	TArray<uint32> UpdatedIds;
```

Comment:
Could you sprinkle these lines with some comments?

Reply:
Get outta town

Reply:
![image](https://user-images.githubusercontent.com/7674163/52272034-50fccf00-293d-11e9-93f4-308c8f6ca83e.png)